paz/backend/boxes.py

```diff
@@ -4,377 +4,552 @@
-    return jp.concatenate([x_min, y_min, x_max, y_max], axis=1)
+    center_x, center_y, W, H = split_into_components(boxes)
+    x_minimum = center_x - (W / 2.0)
+    x_maximum = center_x + (W / 2.0)
+    y_minimum = center_y - (H / 2.0)
+    y_maximum = center_y + (H / 2.0)
+    return jp.concatenate([x_minimum, y_minimum, x_maximum, y_maximum], axis=1)
 
 
 def pad(boxes, size, value=-1):
-    """Pads boxes with given value.
-
-    # Arguments
-        boxes: Array `(num_boxes, 4)`.
-
-    # Returns
-        Padded boxes with shape `(size, 4)`.
-    """
+    """Pad boxes array to specified size with value."""
     num_boxes = len(boxes)
     if num_boxes > size:
         raise ValueError(f"Samples ({num_boxes}) exceeds pad ({size}).")
     padding = ((0, size - num_boxes), (0, 0))
     return jp.pad(boxes, padding, "constant", constant_values=value)
 
 
-def pad_data(boxes, size, value=-1):
-    """Pads list of boxes with a given.
+def order_xyxy(point_A, point_B):
+    """Order coordinates to form a valid bounding box."""
+    x1, y1 = point_A
+    x2, y2 = point_B
+    return min(x1, x2), min(y1, y2), max(x1, x2), max(y1, y2)
 
-    # Arguments
-        boxes: List of size `(num_samples)` containing boxes as lists.
 
-    # Returns
-        boxes with shape `(num_samples, size, 4)`
-    """
-    padded_elements = []
-    for sample in boxes:
-        padded_element = pad(jp.array(sample), size, value)
-        padded_elements.append(padded_element)
-    return jp.array(padded_elements)
+def take_last_two_points(points):
+    """Retrieve the last two points from a list."""
+    return points[-1], points[-2]
 
 
-def flip_left_right(boxes, image_width):
-    """Flips box coordinates from left-to-right and vice-versa.
+def draw_circle_and_store_point(image, points, x, y, radius, color):
+    """Draw a circle on image and store the point."""
+    paz.draw.circle(image, (x, y), radius, color)
+    points.append((x, y))
 
-    # Arguments
-        boxes: Numpy array of shape `(num_boxes, 4)`.
 
-    # Returns
-        Numpy array of shape `(num_boxes, 4)`.
-    """
-    x_min, y_min, x_max, y_max = split(boxes)
-    return join(x_max, y_min, x_min, y_max)
+def process_pair_of_points(image, points, boxes, color, radius):
+    """Process two points to form a box and draw it."""
+    point_A, point_B = take_last_two_points(points)
+    box = order_xyxy(point_A, point_B)
+    paz.draw.box(image, box, color, radius)
+    boxes.append(box)
 
 
-def from_selection(image, radius=5, color=(255, 0, 0), window_name="image"):
-    points, boxes = [], []
-    image = image.copy()
+def on_double_click_event(event, x, y, flags, param, image, points, boxes, radius, color):
+    """Handle double-click event to draw points and boxes."""
+    if event == cv2.EVENT_LBUTTONDBLCLK:
+        draw_circle_and_store_point(image, points, x, y, radius, color)
+        if len(points) % 2 == 0:
+            process_pair_of_points(image, points, boxes, color, radius)
```

Comment:
This functions will most likely not be needed in the public API. Let's define inside the original function "from_selection" all the functions that are explicitly useful for this case. If we ever see that we might need them somewhere else we can put them out.